pyomo/contrib/doe/doe.py

```diff
@@ -1283,14 +1197,14 @@ def trace_calc(m):
             """
             Calculate FIM elements. Can scale each element with 1000 for performance
             """
-            return m.trace == sum(m.fim[j, j] for j in m.regression_parameters)
+            return model.trace == sum(model.fim[j, j] for j in model.parameter_names)
 
-        def det_general(m):
+        def determinant_general(m):
             r"""Calculate determinant. Can be applied to FIM of any size.
             det(A) = \sum_{\sigma in \S_n} (sgn(\sigma) * \Prod_{i=1}^n a_{i,\sigma_i})
             Use permutation() to get permutations, sgn() to get signature
             """
-            r_list = list(range(len(m.regression_parameters)))
+            r_list = list(range(len(model.parameter_names)))
```

Comment:
Using a model object in this rule coming from outside of this function scope. You should use `m` here but remember that you are adding this constraint on a block, `model.obj_cons` so the block is what will be passed in as the first argument.